File: android/src/main/java/com/trustwalletcorereactnative/wallet_core/interfaces/ICoin.kt
package com.trustwalletcorereactnative.wallet_core.interfaces

interface ICoin {
    fun getAddress(mnemonic: String, passphrase: String): String
    fun getSeed(mnemonic: String, passphrase: String): ByteArray
    fun getHexPrivateKey(mnemonic: String, passphrase: String): String
    fun getRawPrivateKey(mnemonic: String, passphrase: String): ByteArray
    fun getHexPublicKey(mnemonic: String, passphrase: String): String
    fun getRawPublicKey(mnemonic: String, passphrase: String): ByteArray
    fun sign(mnemonic: String, passphrase: String, message: String): String
    fun signTransaction(tx: Map<String,Any>,mnemonic: String, passphrase: String): String
    fun signTransaction(tx: Map<String,Any>,privateKey: String): String
//    fun multiSignTransaction(
//        mnemonic: String, passphrase: String,
//        tx: Map<String,Any>,
//        privateKeys: ArrayList<ByteArray>
//    ): ArrayList<ByteArray>
}

File: ios/WalletCore/Protocol/ICoin.swift
//
//  ICoin.swift
//  trust-wallet-core-react-native
//
//  Created by Nguyen Van Toan on 21/5/24.
//

import Foundation

protocol ICoin{
    func getAddress(mnemonic : String, passPharse : String) throws -> String
    
    func getSeed(mnemonic : String , passPharse : String) throws -> Data
    
    func getHexPrivateKey(mnemonic : String, passPharse : String) throws -> String
    
    func getRawPrivateKey(mnemonic : String , passPharse : String) throws -> Data
    
    func getHexPulicKey(mnemonic : String , passPharse : String) throws -> String
    
    func getRawPublicKey(mnemonic : String , passPharse : String) throws -> Data
    
    func sign(mnemonic : String, passPharse : String , message : String) throws -> String
    
    func signTransaction(mnemonic : String, passPharse : String , tx : Dictionary<String,Any>) throws -> String
    
    func signTransaction(privateKey : String , tx : Dictionary<String,Any>) throws -> String
}
